src/main/cpp/src/utils.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+
+namespace spark_rapids_jni {
+
+/**
+ * Represents timestamp with microsecond accuracy.
+ * Max year is six-digits, approximately [-300000, 300000]
+ */
+struct ts_segments {
```

Comment:
We would probably be able to remove this completely and just use the classes from `std::cuda::chrono`.